File: skynda_backend/src/java/me/skynda/classification/controller/ClassificationController.java
package me.skynda.classification.controller;

import me.skynda.classification.dto.ClassificationResponseDto;
import me.skynda.common.controller.BaseController;
import me.skynda.common.interfaces.services.ClassificationService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin(origins = "*")
@RestController
@RequestMapping(value = "/api", produces = "application/json")
public class ClassificationController extends BaseController {

    @Autowired
    private ClassificationService classificationService;

    @RequestMapping(value = "/classifications/{type}", method = RequestMethod.GET)
    public List<ClassificationResponseDto> getAll(@PathVariable String type) {
        return classificationService.getByType(type);
    }

    @RequestMapping(value = "/classifications/{type}/vehicle-bound", method = RequestMethod.GET)
    public List<ClassificationResponseDto> getVehicleBound(@PathVariable String type) {
        return classificationService.getByTypeAndVehicleBound(type);
    }

}

File: skynda_dotnet_backend/Triven.Api/Controllers/ClassificationController.cs
using System.Web.Http;
using Triven.Application;
using Triven.Domain.Services;

namespace Triven.API.Controllers
{
    [RoutePrefix("api/classification")]    // TODO: use singular, not plural
    public class ClassificationController : BaseController
    {
        private readonly IClassificationService _service;

        public ClassificationController()
        {
            _service = IoC.Get<IClassificationService>();
        }

        [HttpGet, Route("{type}")]
        public IHttpActionResult GetAll([FromUri] string type)
        {
            return HandleResult(_service.GetByType(type));
        }

        [HttpGet, Route("{type}/vehicle-bound")]
        public IHttpActionResult GetVehicleBound([FromUri] string type) => HandleResult(_service.GetByTypeAndVehicleBound(type));
    }
}